torchvision/models/vgg.py

```diff
@@ -60,31 +66,31 @@ def _initialize_weights(self):
                 nn.init.constant_(m.bias, 0)
 
 
-def make_layers(cfg, batch_norm=False):
+def make_layers(cfg: List[Union[str, int]], batch_norm: bool = False) -> nn.Sequential:
     layers = []
     in_channels = 3
     for v in cfg:
-        if v == 'M':
-            layers += [nn.MaxPool2d(kernel_size=2, stride=2)]
-        else:
+        if isinstance(v, int):
```

Comment:
FYI: while this doesn't change the code for the default cases, some background on why this was implemented as it was:
if you have the `cfg` as a command-line argument, you will get a list of strings out of the parser (because you have at least one string, so everything should be a string).

I don't think that anyone was using this in this way in this code, so this change is fine, but it could potentially break one workflows.

Reply:
I think the problem here is that if `v` is `Union[int, str]` there is no way of knowing that `v` is only `int` in the `else` branch of `if v == "M"`. I think that could be solved by including a `v = cast(int, v)` into that branch.

Reply:
yes, adding a `cast(int, v)` could be a solution.

Reply:
Done!